woocommerce/woocommerce-ios
# Introduce order details address validation analytics

Closes: #7546

### Summary
This PR validates the shipping address in the order details screen and tracks the information when the address contains errors. Our goal is to enhance the user experience by adding an address validation mechanism similar to the one the shipping labels plugin uses, but first, we want to understand better the impact these changes could have.

The implementation strategy follows the code introduced by the `ShippingLabelAddressFormViewModel`, but in a reusable class format that's only focused on validating an address locally and remotely, without the specifics introduced in the shipping label feature. 

Please bear in mind that this is temporary code to achieve user measurements and understand some behaviors. It's possible to make some changes in the future so the `ShippingLabelAddressFormViewModel` could take advantage of the same code, avoid repetition, and improve code usage. Still, I think that should be tackled later in a separate PR since adjusting the current implementation of the `ShippingLabelAddressFormViewModel` to the code introduced here would be no simple task. 

### How to test

TC1
1. Open the order list
2. Select an order containing a shipping address with errors (invalid country/state names, empty Address 1 strings, etc)
3. Check that the errors are tracked in an event similar to this one:

```
🔵 Tracked: order_address_validation_error, Properties: {"error_message":"The phone number is invalid","validation_scenario":"local","order_id":683,"blog_id":202934350,"is_wpcom_store":true,"is_debug":true}
```

TC2
1. Open the order list
2. Select an order without a shipping address
3. Check that no errors are tracked

TC3
1. Deactivate the WooCommerce Shipping & Tax plugin in your store
2. Restart the Woo app on your device
3. Open the order list
4. Select an order
5. Check that the remote validation is not sent when the plugin is not activated (No request is sent to `/wc/v1/connect/normalize-address/&_method=post`) 

You can use Flipper to check TC3 and verify a request, as the one below never happens
<img width="1024" alt="Screen Shot 2022-08-28 at 14 07 45" src="https://user-images.githubusercontent.com/18119390/187086235-1fa731ff-a584-4d82-aaee-d7370994a15f.png">
